### PR TITLE
Java: Add Java v1 core draft

### PR DESCRIPTION
**Pull request by milderhc**
### Contribution Checklist

- [ ] The code builds clean without any errors or warnings
- [ ] The PR follows the [SK Contribution Guidelines](https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md) and the [pre-submission formatting script](https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md#development-scripts) raises no violations
- [ ] All unit tests pass, and I have added new tests where possible
- [ ] I didn't break anyone :smile:

### CODE REVIEW

File: java/semantickernel-core/src/main/java/com/microsoft/semantickernel/functions/SemanticFunctionResult.java [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+        return null;
+    }
+
+    @Override
+    public Flux<String> getStreamingValueAsync() {
+        return streamingValueAsync;
+    }
+
+    @Override
+    public Mono<String> getValueAsync() {
```
Comment (dsgrieve) [inline java/semantickernel-core/src/main/java/com/microsoft/semantickernel/functions/SemanticFunctionResult.java:58]: I think the value should always be a Flux. getValueAsync would convert the Flux to a Mono.

File: java/semantickernel-api/src/main/java/com/microsoft/semantickernel/orchestration/FunctionResult.java [hunk 2]
```diff
@@ -1,53 +1,54 @@
...
     Map<String, Object> metadata();
 
     /**
      * Get the function result.
+     *
      * @return The function result.
-     * @param <T> The type of the function result.
      */
-    <T> Mono<T> getValueAsync();
+    Mono<String> getValueAsync();
```
[the remark this answers is not in the document]
Comment (milderhc) [reply java/semantickernel-api/src/main/java/com/microsoft/semantickernel/orchestration/FunctionResult.java:39]: Right, I changed it to this based on what the API response usually is, they actually return Mono<String> and Flux<String>, also .NET at the end use strings: https://github.com/matthewbolanos/sk-v1-proposal/blob/b4e7310a0cfd6abb23c1e1a457f18509b2d9fcb5/dotnet/src/extensions/SKHandleBars/AIServices/AzureOpenAI/AzureOpenAIChatCompletion.cs#L65. 

Do you want me to change back this on the API?

File: java/extensions/semantickernel-planners/src/main/java/com/microsoft/semantickernel/planner/actionplanner/Plan.java [hunk 3]
```diff
@@ -30,7 +30,7 @@
 import reactor.core.publisher.Mono;
 
 /** Standard Semantic Kernel callable plan. Plan is used to create trees of SKFunctions. */
-public class Plan extends AbstractSkFunction<CompletionRequestSettings> {
+public class Plan extends AbstractSkFunction {
```
[the remark this answers is not in the document]
Comment (milderhc) [reply java/extensions/semantickernel-planners/src/main/java/com/microsoft/semantickernel/planner/actionplanner/Plan.java:33]: Right!

File: java/semantickernel-core/src/main/java/com/microsoft/semantickernel/functions/SemanticFunctionResult.java (hunk 1 above)
Comment (milderhc) [reply java/semantickernel-core/src/main/java/com/microsoft/semantickernel/functions/SemanticFunctionResult.java:58]: The API returns a Mono<String> for non-streaming and Flux<String> for streaming. Wouldn't be converting back and forth Mono->Flux->Mono when using non-streaming completion?

File: java/semantickernel-core/src/main/java/com/microsoft/semantickernel/semanticfunctions/SemanticFunction.java [hunk 4]
```diff
@@ -0,0 +1,211 @@
...
+import java.io.InputStream;
+import java.util.Collections;
+import java.util.List;
+import javax.annotation.Nullable;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+
+public class SemanticFunction extends DefaultSemanticSKFunction {
```
[the remark this answers is not in the document]
Comment (milderhc) [reply java/semantickernel-core/src/main/java/com/microsoft/semantickernel/semanticfunctions/SemanticFunction.java:26]: Right, that breaks it. Although, SemanticFunction is a form/implementation of SKFunction and not all SKFunction would have the ability to load from a yaml file. Should we still add it? Or add SemanticFunction to the api module and then implement another class in the core module?

**Pull request merged**